src/ecdsa/keys.py

```diff
@@ -923,8 +947,48 @@ def from_der(cls, string, hashfunc=sha1):
         :return: Initialised SigningKey object
         :rtype: SigningKey
         """
-        string = normalise_bytes(string)
-        s, empty = der.remove_sequence(string)
+        s = normalise_bytes(string)
+        curve = None
+
+        # PKCS #8 has the algorithm identifier, including the curve name,
+        # before the actual key. Then it contains the key data within an
+        # octet string.
+        if pkcs8:
+            s, empty = der.remove_sequence(s)
+            if empty != b(""):
+                raise der.UnexpectedDER(
+                    "trailing junk after DER privkey: %s"
+                    % binascii.hexlify(empty)
+                )
+
+            version, s = der.remove_integer(s)
+            if version != 0 and version != 1:
+                raise der.UnexpectedDER(
+                    "expected '0' or '1' at start of DER privkey, got %d"
```

Comment:
why not say that the encoded version is unsupported?

Reply:
Not sure what you mean here... the version only indicates whether the public key is present in the top-level structure.

Reply:
but the element of the structure is called "version", so we should tell the user that the decoded "version" is not one we expect, and it's X